kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -201,4 +260,35 @@ private void validate(String value) {
     private static void addConfig(HashMap<String, TableConfig<?>> configs, TableConfig<?> config) {
         configs.put(config.getKey().toLowerCase(Locale.ROOT), config);
     }
+
+    private static Map<String, String> parseStringToJsonMap(String jsonString) {
```

Comment:
Could add some tests?
1. A few happy path cases
2. Handling of escaped characters
3. Test the two areas where you throw exception
I would recommend creating some complex static json with unicode characters using a standard serializer and then deserializing the static json in a test.

Reply:
yep, and this can be its own separate PR with the design decision doc attached to it.